client/src/panels/panel_declarations/finances/detailed_program_spending_split/detailed_program_spending_split.js

```diff
@@ -124,6 +125,10 @@ class HistoricalProgramBars extends React.Component {
                   selected: _.toggle_list(selected, id),
                 });
               }}
+              onToggleClick={() => this.setState({
+                selected: selected.length < all_keys.length ? all_keys : [],
```

Comment:
I believe all our legends usually stop the user from toggling off the final item. I don't remember if this was because graphs broke when they had no data or if we just didn't want them to be able to (since they would at least *look* broken). Is being able to have no selected items valuable? Does that clash with the normal behaviour too much?